### PR TITLE
use contextvar instead of threadlocal

### PR DESCRIPTION
**Pull request by hamedsh**
replace threading.local() with contextvars to be able to use with Django 4.2

### CODE REVIEW

File: auditlog/context.py [hunk 1]
```diff
@@ -48,26 +48,30 @@ def _set_actor(user, sender, instance, signal_duid, **kwargs):
     This function becomes a valid signal receiver when it is curried with the actor and a dispatch id.
     """
     try:
-        auditlog = threadlocal.auditlog
-    except AttributeError:
+        auditlog = auditlog_value.get()
+    except LookupError as ex:
```
Comment (hramezani) [inline auditlog/context.py:52]:
```suggestion
    except LookupError:
```

**Comment by hramezani** [comment]
Thanks @hamedsh for this. I've added a commit to fix the lint error.

Let's see what are @aqeelat and @aleh-rymasheuski ideas here.

**Review by aleh-rymasheuski — COMMENTED** [review]
Not familiar with the API so can't review the code properly, but this change seems logical.

My only concern is that the `threadlocal` name is public (though undocumented) API so we should (a) bump the "feature" part of the version number (or squeeze this change into the v3 major release) and (b) document this change in the changelog.

**Comment by hramezani** [comment]
We will release it in V3.

Yeah, the change has to be mentioned in changelog

**Comment by hramezani** [comment]
@hamedsh Could you please add an entry to changelog?

**Review by hramezani — APPROVED** [review]

**Pull request merged**